src/Bugsnag.Unity/Payload/App.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using AOT;
+using UnityEngine;
+
+namespace Bugsnag.Unity.Payload
+{
+  /// <summary>
+  /// Represents the "app" key in the error report payload.
+  /// </summary>
+  class App : Dictionary<string, object>, IFilterable
+  {
+    internal App(IConfiguration configuration) : this(configuration.AppVersion, configuration.ReleaseStage, null)
+    {
+
+    }
+
+    internal App(string version, string releaseStage, string type)
+    {
+      this.AddToPayload("version", version);
+      this.AddToPayload("releaseStage", releaseStage);
+      this.AddToPayload("type", type);
+    }
+  }
+
+  class AndroidApp : App
+  {
+    internal AndroidApp(IConfiguration configuration, AndroidJavaObject client) : base(configuration)
+    {
+      using (var appData = client.Call<AndroidJavaObject>("getAppData"))
+      using (var map = appData.Call<AndroidJavaObject>("getAppData"))
+      {
+        this.PopulateDictionaryFromAndroidData(map);
+      }
+    }
+  }
+
+  class MacOsApp : App
+  {
+    [DllImport("bugsnag-osx", EntryPoint = "retrieveAppData")]
+    static extern void RetrieveAppData(IntPtr instance, Action<IntPtr, string, string> populate);
+
+    internal MacOsApp(IConfiguration configuration) : base(configuration)
+    {
+      GCHandle handle;
+
+      try
+      {
+        handle = GCHandle.Alloc(this);
```

Comment:
This bit doesn't build on windows, as it complains that the `Free()` call is possible invoked on an uninitialized `handle` variable.

I changed it to:

```cs
      GCHandle handle;

      try
      {
        handle = GCHandle.Alloc(this);

        try
        {
          RetrieveAppData(GCHandle.ToIntPtr(handle), PopulateAppData);
        }
        finally
        {
          handle.Free();
        }
      }
      catch (ArgumentException _ex)
      {
        
      }
```

Though there might be a better way of doing this.